### PR TITLE
Allocate value and destroy it via new/delete cpp commands

### PR DESCRIPTION
**Pull request by Takishipp**

### CODE REVIEW

**Comment by Takishipp** [comment]
I couldn't change all c allocations to cpp allocations.
Do you'v some ideas?

File: include/xbt/sysdep.h [hunk 1]
```diff
@@ -113,6 +113,8 @@ static XBT_ALWAYS_INLINE void *xbt_realloc(void *p, size_t s) {
     @hideinitializer */
 #define xbt_free(p) free(p) /*nothing specific to do here. A poor valgrind replacement? */
 
+#define xbt_delete(p) (delete[] p) /*nothing specific to do here. A poor valgrind replacement? */
```
Comment (mquinson) [inline include/xbt/sysdep.h]: There is no need for this macro. Write delete[] directly where you've called this macro, instead.

File: include/xbt/sysdep.h [hunk 2]
```diff
@@ -125,6 +127,8 @@ SG_END_DECL()
     @hideinitializer */
 #define xbt_new0(type, count) ((type*)xbt_malloc0 (sizeof (type) * (count)))
 
+#define xbt_new1(type, count) (new type[count])
```
Comment (mquinson) [inline include/xbt/sysdep.h]: There is no need for this  macro either. Please inline it as well.

File: src/instr/instr_paje_values.cpp [hunk 3]
```diff
@@ -15,7 +15,7 @@ value::value(const char* name, const char* color, type_t father)
   if (name == nullptr || father == nullptr){
     THROWF (tracing_error, 0, "can't create a value with a nullptr name (or a nullptr father)");
   }
-  this->ret         = xbt_new0(value, 1);
+  this->ret         = xbt_new1(value, 1);
```
Comment (mquinson) [inline src/instr/instr_paje_values.cpp]: This is where the important thing should take place. Remove this new, and kill the definition of this->ret.

File: src/instr/instr_paje_values.cpp [hunk 4]
```diff
@@ -45,8 +41,9 @@ value* value::get_or_new(const char* name, const char* color, type_t father)
     ret = value::get(name, father);
   }
   catch(xbt_ex& e) {
-    value rett(name, color, father);
-    ret = rett.ret;
+    ret = new value(name, color, father);
+    ret = ret->ret;
```
Comment (mquinson) [inline src/instr/instr_paje_values.cpp]: once the ret field is gone, this line is useless.

File: src/instr/instr_smpi.h [hunk 5]
```diff
@@ -29,7 +29,6 @@ XBT_PRIVATE void TRACE_smpi_testing_out(int rank);
 XBT_PRIVATE void TRACE_smpi_testing_in(int rank, instr_extra_data extra);
 XBT_PRIVATE void TRACE_smpi_alloc();
 XBT_PRIVATE void TRACE_smpi_release();
-XBT_PRIVATE void TRACE_smpi_ptp_in(int rank, const char* operation, instr_extra_data extra);
```
Comment (mquinson) [inline src/instr/instr_smpi.h]: This is a completely unrelated change, so don't put this in the same pull request.

Moreover, I already said in the previous pull request that this change was not right. I don't want to merge the ptp_in signal with the collective_in signal.

File: src/smpi/bindings/smpi_pmpi_request.cpp [hunk 6]
```diff
@@ -170,7 +170,7 @@ int PMPI_Irecv(void *buf, int count, MPI_Datatype datatype, int src, int tag, MP
     if(known==0)
       dt_size_send = datatype->size();
     extra->send_size = count*dt_size_send;
-    TRACE_smpi_ptp_in(rank, __FUNCTION__, extra);
```
Comment (mquinson) [inline src/smpi/bindings/smpi_pmpi_request.cpp]: Remove this from the PR.

File: src/smpi/bindings/smpi_pmpi_request.cpp [hunk 7]
```diff
@@ -218,7 +218,7 @@ int PMPI_Isend(void *buf, int count, MPI_Datatype datatype, int dst, int tag, MP
...
```
Comment (mquinson) [inline src/smpi/bindings/smpi_pmpi_request.cpp]: Remove this from the PR.

File: src/smpi/bindings/smpi_pmpi_request.cpp [hunk 8]
```diff
@@ -266,7 +266,7 @@ int PMPI_Issend(void* buf, int count, MPI_Datatype datatype, int dst, int tag, M
...
```
Comment (mquinson) [inline src/smpi/bindings/smpi_pmpi_request.cpp]: Remove this from the PR.

File: src/smpi/internals/instr_smpi.cpp [hunk 9]
```diff
@@ -372,22 +372,6 @@ void TRACE_smpi_testing_out(int rank)
   new PopStateEvent (SIMIX_get_clock(), container, type);
 }
 
-void TRACE_smpi_ptp_in(int rank, const char *operation, instr_extra_data extra)
```
Comment (mquinson) [inline src/smpi/internals/instr_smpi.cpp]: Remove this from the PR.

**Comment by mquinson** [comment]
Your changes of today did not improve the situation. Not at all. Please redo your pull request with only the right commits.
